### PR TITLE
UCT/IB/MLX5: fixed compilation issue

### PR DESCRIPTION
**Pull request by hoopoepg**
- fixed compilation on ARM platforms

### CODE REVIEW

File: src/uct/ib/mlx5/ib_mlx5.inl [hunk 1]
```diff
@@ -113,12 +113,14 @@ uct_ib_mlx5_inline_copy(void *restrict dest, const void *restrict src, unsigned
 {
     ptrdiff_t n;
 
-    if (dest + length <= wq->qend) {
-        memcpy(dest, src, length);
-    } else {
-        n = wq->qend - dest;
-        memcpy(dest, src, n);
-        memcpy(wq->qstart, src + n, length - n);
+    if (length) {
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.inl]: looks like extra branch on fast-path? can't we silence the compiler somehow or make it only for ARM?

Comment (hoopoepg) [reply src/uct/ib/mlx5/ib_mlx5.inl]: below there are conditions where length used, new condition should not affect to performance.

and original code is not valid: memcpy has unpredictable behavior on NULL ptr in zero-length.

so, look for ARM specific suppression?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6403/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
looks like gcc has hard time to propagate constants, this happens from uct_rc/dc_mlx5_ep_tag_eager_zcopy() only. where the inline copy function is not called.
so can pass some non-NULL pointer as "am_header" just to silence to compiler, for example "iface", "ep", etc.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6404/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6406/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6408/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe tests are passed

**Pull request merged**